pkg/apis/extensions/validation/operatingsystemconfig.go

```diff
@@ -122,8 +123,14 @@ func ValidateFiles(files []extensionsv1alpha1.File, fldPath *field.Path) field.E
 				allErrs = append(allErrs, field.Required(idxPath.Child("content", "secretRef", "dataKey"), "field is required"))
 			}
 		case file.Content.Inline != nil:
-			if len(file.Content.Inline.Encoding) == 0 {
-				allErrs = append(allErrs, field.Required(idxPath.Child("content", "inline", "encoding"), "field is required"))
+			switch file.Content.Inline.Encoding {
+			case
+				"",
+				string(cloudinit.B64FileCodecID),
+				string(cloudinit.GZIPFileCodecID),
+				string(cloudinit.GZIPB64FileCodecID):
+			default:
+				allErrs = append(allErrs, field.Required(idxPath.Child("content", "inline", "encoding"), "field is required to among b64, gzip or gzip+b64"))
```

Comment:
The error here should not be `Required`, but `NotSupported`, see:
```
// NotSupported returns a *Error indicating "unsupported value".
// This is used to report unknown values for enumerated fields (e.g. a list of
// valid values).
```
The details should include a list of all supported values. Checking if a string is in a list of valid values can be done easily with `utils.ValueExists`.

Reply:
This change breaks the `#ValidOperatingSystemConfig` so the test should be adapted as well. Use `make verify` before amending pull requests to make sure that unit and integration tests pass.